src/gamemode.c

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright © 2019 Red Hat, Inc
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Christian J. Kellner <christian@kellner.me>
+ */
+
+#include "config.h"
+
+#include "request.h"
+#include "permissions.h"
+
+#include "xdp-dbus.h"
+#include "xdp-impl-dbus.h"
+#include "xdp-utils.h"
+
+#include <gio/gunixfdlist.h>
+
+#define _GNU_SOURCE 1
+#include <errno.h>
+#include <locale.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/un.h>
+#include <sys/wait.h>
+#include <unistd.h> /* unlinkat, fork */
+
+/* well known names*/
+#define GAMEMODE_DBUS_NAME "com.feralinteractive.GameMode"
+#define GAMEMODE_DBUS_IFACE "com.feralinteractive.GameMode"
+#define GAMEMODE_DBUS_PATH "/com/feralinteractive/GameMode"
+
+#define PERMISSION_TABLE "gamemode"
+#define PERMISSION_ID "gamemode"
+
+/* */
+typedef struct _GameMode GameMode;
+typedef struct _GameModeClass GameModeClass;
+
+static gboolean handle_query_status (XdpGameMode *object,
+                                     GDBusMethodInvocation *invocation,
+                                     gint pid);
+
+static gboolean handle_register_game (XdpGameMode *object,
+                                      GDBusMethodInvocation *invocation,
+                                      gint pid);
+
+static  gboolean handle_unregister_game (XdpGameMode *object,
+                                         GDBusMethodInvocation *invocation,
+                                         gint pid);
+
+/* globals  */
+static GameMode *gamemode;
+
+/* gobject  */
+
+struct _GameMode
+{
+  XdpGameModeSkeleton parent_instance;
+
+  /*  */
+  GDBusProxy *client;
+};
+
+struct _GameModeClass
+{
+  XdpGameModeSkeletonClass parent_class;
+};
+
+GType gamemode_get_type (void) G_GNUC_CONST;
+static void gamemode_iface_init (XdpGameModeIface *iface);
+
+G_DEFINE_TYPE_WITH_CODE (GameMode, gamemode, XDP_TYPE_GAME_MODE_SKELETON,
```

Comment:
Given that you have a capital M here, I would expect the prefix to be game_mode

Reply:
True, I changed it to `game_mode`.